Completion/Unix/Command/_code

```diff
@@ -0,0 +1,42 @@
+#compdef code
+
+local arguments
+
+__code_locales() {
+  local -a locales
+  locales=($(locale -a | sed -e 's/\..*$//'))
+  _describe 'locale' locales
+}
+
+arguments=(
+  {-d,--diff}'[compare two files with each other]:file to compare:_files:file to compare with:_files'
+  {-a,--add}'[add folder(s) to the last active window]:folder:_directories'
+  {-g,--goto}'[open a file at the path on the specified]'
+  {-n,--new-window}'[force to open a new window]'
+  {-r,--reuse-window}'[force to open a file or folder in the last active window]'
```

Comment:
Are these last two options mutually exclusive. If so, we can indicate to zsh that only one should be used. It is also usually the case that long and short option forms are mutually exclusive of each other. Furthermore, the English in these descriptions is not great. I know it comes from code's --help output but the word "force" when used as a verb like this should have an object. So, I'd suggest something like:

    '(-n --new-window -r --reuse-window)'{-n,--new-window}'[open a new window]'

And similarly for -r.

Reply:
Good point. I followed your advice regarding -r and -n. I have also made all long and short arguments exclusive